### PR TITLE
Expand `web-tests` application for `client-js` tests

### PR DESCRIPTION
**Pull request by YegorUdovchenko**
### Summary
This PR brings updates to the project integration tests allowing to test `spine-web` library artifact against the emulated backend server. Previously, tests of `client-js` module used the test application deployed at AppEngine which is no longer needed. Now all tests are bounded within project.

### Integration tests
`./integration-tests` package contains modules for testing of the unpublished `spine-web` NPM library against the Spine-based server. The `test-app` module contains a simplified Spine application. The `web-tests` module defines tests acting as a client-side application where client-server interactions are carried out by `spine-web` NPM library.

### `test-app`
The `test-app` module contains a simplified Spine application. The `web-tests` module uses this
application as a test backend. The application uses the Spine `web` API (with the `firebase-web`
library). The application model contains a minimal set of commands, events, and projections that
allows to test functionality of the `spine-web` library. The application uses Firebase application emulated locally with a [`firebase-server`](https://www.npmjs.com/package/firebase-server) tool. This tool is executed from the `web-tests` module `node_modules` folder.

#### Model changes
The `TaskItem` projection was created. It duplicates state of the respective aggregate and expands it with some columns available for filtering;

### `web-tests`
The `web-tests` module performs tests of the unpublished artifact of the `spine-web` library. It
defines tests acting as a client-side application that uses `spine-web` library for interactions
with a backend. Performs tests using the unpublished artifact of the `spine-web` library. This approach allows to test not only the library functionality but also ensure it is properly built.

### `client-js`
 - the dependencies of the `client-js` module were upgraded to address https://github.com/SpineEventEngine/web/issues/88;
 - tests of the Firebase client were moved to the `web-tests` module;

### CODE REVIEW

**Comment by YegorUdovchenko** [comment]
@armiol PTAL

File: client-js/test/client/firebase-client/command-test.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    CommandHandlingError,
+    ConnectionError
+} from '@lib/client/errors';
+import {fail} from '../test-helpers';
+import {client, initClient} from './given/firebase-client';
+
+describe('FirebaseClient command sending', function () {
+
+    // Big timeout due to remote calls during tests.
+    const timeoutDuration = new Duration({minutes: 2});
```
Comment (armiol) [inline client-js/test/client/firebase-client/command-test.js]: Let's discuss. I don't think we should run tests against the remote API. We have tried to do so, it's unstable.

**Comment by YegorUdovchenko** [comment]
@armiol PTAL

**Comment by YegorUdovchenko** [comment]
@armiol PTAL

File: .travis.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ env:
 before_install:
   - chmod +x gradlew
 
+
```
Comment (armiol) [inline .travis.yml]: Please kill this line.

File: integration-tests/test-app/src/main/proto/spine/web/test/given/commands.proto [hunk 3]
```diff
@@ -29,6 +32,13 @@ message RenameTask {
     string name = 2 [(required) = true];
 }
 
+message ReassignTask {
+
+    TaskId id = 1;
+
+    spine.core.UserId assignee = 2 [(required) = true];
```
Comment (armiol) [inline integration-tests/test-app/src/main/proto/spine/web/test/given/commands.proto]: Is this `new_assignee`?

File: integration-tests/web-tests/test/test-helpers.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * Can be used in callback-based async tests to fail them before waiting
```
Comment (armiol) [inline integration-tests/web-tests/test/test-helpers.js]: As the first sentence, please say what it is, and then how it can be used.

**Pull request merged**